src/std/text/json.ss

```diff
@@ -92,19 +92,23 @@ package: std/text
     (let lp ()
       (let (key (read-json-hash-key port))
         (if key
-          (let (val (read-json-object port))
-            (hash-put! obj key val)
-            (skip-whitespace port)
-            (let (char (peek-char port))
-              (case char
-                ((#\,)
-                 (read-char port)
-                 (lp))
-                ((#\})
-                 (read-char port)
-                 obj)
-                (else
-                 (raise-invalid-token port char)))))
+          (begin
```

Comment:
I would rather write this without a begin, but with an if instead.

That is, instead of `(begin (when ...) (let ...))` write it as:
```
(if (hash-key? obj key)
   (error ...)
   (let ...))
```